disnake/enums.py

```diff
@@ -821,6 +822,253 @@ class ThreadLayout(Enum):
     gallery_view = 2
 
 
+class ClientEvent(str, Enum):
```

Comment:
maybe we could support both the grouping and the general `Events` class

```py
from enum import Enum

class ClientEvent(Enum):
	ready = "ready"
	connect = "connect"

class test(type):
	def __new__(cls, name, bases, dct):
		x = super().__new__(cls, name, bases, dct)
		for i in ClientEvent:
			setattr(x, i.value, i.value)
		return x

class Events(metaclass=test):
	pass
```
```py
>>> Events.ready
"ready"
>>> Events.connect
"connect"
```

I'm aware that this could be improved but this is what I thought we could do

Reply:
Something like that won't behave well with static type-checkers :/

The only thing I can imagine working here, to have both grouped events and one general `Events` type, would be something like:
```py
class ClientEvent(Enum):
    ready = "ready"

class Event:  # note: no base type
    ready = ClientEvent.ready
```
This would be type-checker/linter friendly and works at runtime, but is essentially just a namespace. It'd also require a lot of duplication, either through manual work (:notlikethis:) or codegen, and would make documentation difficult.
There are a couple other options, like not using enums at all, but they all ultimately fail in some way once you run pyright/mypy :confused: 

In the end, a combined option is probably not viable, we'd have to choose one of the two options; I'm leaning towards using a single enum class, since many events are already somewhat categorized just by their prefix (thread events all look like `thread_xyz`), and there's always documentation to help out.


(linking related conversation here for future reference: https://canary.discord.com/channels/808030843078836254/913779868985090089/1062473728656687164)